src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -1872,10 +2083,10 @@ private void doEncodeConnack(
         {
             int propertiesSize = 0;
 
-            if (sessionExpiryInterval > 0)
+            if (sessionExpiryIntervalLimit > 0)
             {
                 mqttPropertyRW.wrap(propertyBuffer, propertiesSize, propertyBuffer.capacity())
-                              .sessionExpiry(sessionExpiryInterval)
+                              .sessionExpiry(sessionExpiryIntervalLimit)
```

Comment:
Assign `wrap(...)` return value to local `mqttProperty` variable instead of using `mqttPropertyRW.limit()`.